### ISSUE TITLE
Enable support for non-tty

### ISSUE THREAD

**Issue by cdrage** [projectatomic/atomic#279]
We currently have an issue of running atomic cli in a jenkins-like environment where no tty is present (https://github.com/projectatomic/atomicapp/issues/430)

What we get when running an example within Jenkins CI is:
```
...
11:34:12 docker run -it --rm --privileged -v /home/jenkins/workspace/ ...
11:34:12 cannot enable tty mode on non tty input
```

Which relates to the usage of using tty when running docker run. An example of a solution would be to simply pass the `--tty=false` to the docker run command in order to allocate a pseudo-tty. Similar to the code in https://github.com/projectatomic/atomic/blob/70612963daf86097fc2e895cda544dc5b094113c/Atomic/top.py we can allocate an if statement if no tty is present using the `tty` library.

ex from `top.py`
```python 
        # Activate optional columns
        self._activate_optionals()
        # Do we have a tty?
        # Can run ./atomic top <&- to replicate no tty
        has_tty = isatty(0)
        # Set up terminal, input handling
        if has_tty:
            fd = sys.stdin.fileno()
            old_settings = termios.tcgetattr(fd)
```

Thoughts? :) Would be a good use-case for non-tty deployments (for ex. ssh'ing in to an atomic machine and running atomic cli without a tty)

**Comment by rhatdan** [comment]
@baude PTAL

**Comment by jlebon** [comment]
One issue is that e.g. containers which do require a TTY will now fail/crash. So in that case, it just moves the failure point from docker to the app.

The advantage of passing -t or not -t ourselves is that it gives container developers the flexibility to act how they want in either situation (e.g. with -t, give a prompt or live feed, but without -t, just start the daemon). The disadvantage is that we're potentially disregarding their wishes (e.g. they might be coding with the assumption that they'll have a TTY since they explicitly wrote `-t` in the RUN label). I guess that can be offset by good documentation.

**Comment by cdrage** [comment]
Closing this as we're going to fix this atomicapp side. :) Too small of a niche for atomic CLI.

**Comment by jlebon** [comment]
Just a note, I didn't mention this issue in the commit messages, but the concerns raised here should be fixed by #776, which only adds `-t` to the default args if a TTY is present. Though note that we don't inspect and try to edit what's in the `RUN` label, which I think is the right thing to do, as mentioned earlier.
